lib/appnexusapi/service.rb

```diff
@@ -54,6 +55,19 @@ def get(params={})
     end
   end
 
+  def get_all(params = {})
+    responses = []
+    last_responses = get(params)
+
+    while last_responses.size > 0
+      responses += last_responses
+      last_responses = get(params.merge('start_element' => responses.size))
+      sleep(1) # The gem has no error handling at all for rate limit errors; sleeping for a second prevents errors
```

Comment:
to reiterate - the real solution would probably be to catch rate limit errors in some faraday middleware that automatically waits and retries and not have any timeout here... but that would require massive changes to the gem that i'm not really prepared to make.